analyses/fusion_filtering/04-project-specific-filtering.Rmd

```diff
@@ -0,0 +1,262 @@
+    label: "Grouping variable"
+    value: broad_histology
+    input: string
+  dataStranded:
+    label: "Input filtered fusion dataframe"
+    value: scratch/standardFusionStrandedExp_QC_expression_GTExComparison_annotated.RDS
+    input: file
+  dataPolya:
+    label: "Input filtered fusion dataframe"
+    value: scratch/standardFusionPolyaExp_QC_expression_GTExComparison_annotated.RDS
+    input: file    
+  numCaller:
+    label: "Least Number of callers to have called fusion"
+    value: 2
+    input: integer
+  numSample:
+    label: "Least Number of samples to have fusion per group"
+    value: 2
+    input: integer
+  numGroup:
+    label: "Max number of groups found in"
+    value: 1
+    input: integer
+  limitMultiFused:
+    label: "Max number of times gene can be fused per sample"
+    value: 5
+    input: integer  
+  outputfolder:
+    label: "results folder for *tsv files"
+    value: results
+    input: string     
+
+    
+---
+
+K S Gaonkar
+
+Filtered Fusions:
+1. In-frame fusions is called in atleast 2 samples per histology OR
+2. In-frame fusions is called in atleast 2 callers 
+AND
+Filtered-fusions found in more than 1 histology OR
+Filtered-fusion doesn't have multi-fused gene (more than 5 times in sample)
+
+Putative Driver:
+Filtering for general cancer specific genes
+Fusions with genes in either onco
+
+This notebook assumes you are in OpenPBTA-analysis project folder structure.
+
+
+```{r}
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+setwd(root_dir)
+#load required packages
+suppressPackageStartupMessages(library("readr"))
+suppressPackageStartupMessages(library("tidyverse"))
+suppressPackageStartupMessages(library("reshape2"))
+suppressPackageStartupMessages(library("qdapRegex"))
+
+#read filtFusion files
+strandedQCGeneFiltered_filtFusion<-readRDS(params$dataStranded)
+polyaQCGeneFiltered_filtFusion<-readRDS(params$dataPolya)
+
+QCGeneFiltered_filtFusion<-rbind(strandedQCGeneFiltered_filtFusion,polyaQCGeneFiltered_filtFusion)
+
+write.table(QCGeneFiltered_filtFusion,"analyses/fusion_filtering/results/FilteredFusion.tsv",sep="\t",quote=FALSE,row.names = FALSE)
+
+# subset for recurrent fusion detection and multifused genes QC
+fusion_calls<-unique(QCGeneFiltered_filtFusion)
+# remove distance from intergenic fusions
+fusion_calls$FusionName<-unlist(lapply(fusion_calls$FusionName,function(x) rm_between(x, "(", ")", extract = F)))
+
+
+# get grouping column id
+group<-params$group
+
+# get histology file
+clinical<-read.delim(params$histology,stringsAsFactors = FALSE)
+clinical<-clinical[,c("Kids_First_Biospecimen_ID","Kids_First_Participant_ID","broad_histology")]
+
+# Least number of callers
+numCaller<-params$numCaller
+
+# Least number of samples per group
+numSample<-params$numSample
+
+# Max number of groups 
+numGroup<-params$numGroup
+
+# Max number of times gene can be fused per sample
+limitMultiFused<-params$limitMultiFused
+
+# results folder
+outputfolder<-params$outputfolder
```

Comment:
Move this up to before when you write `QCGeneFiltered_filtFusion` to file.

Reply:
done!